src/app/pages/storage/volumes/manager/manager.component.ts

```diff
@@ -570,51 +524,61 @@ export class ManagerComponent implements OnInit, OnDestroy, AfterViewInit {
+            layout.push({ type: vdev.type.toUpperCase(), disks: disks });
           }
         });
 
         let body = {};
-        this.loader.open();
         if (this.isNew) {
-          body = {volume_name: this.name, encryption: this.isEncrypted, layout: layout };
+          body = {name: this.name, encryption: this.isEncrypted, topology: { data: layout } };
         } else {
-          body  = {volume_add: this.name, layout: layout };
+          body  = {name: this.name, topology: layout };
         }
-        this.busy =
-          this.rest
-          .post(this.resource_name, {
-            body: JSON.stringify(body)
-          })
-          .subscribe(
-            (res) => {
-              this.loader.close();
-              if(this.isEncrypted) {
-                let dialogRef = this.mdDialog.open(DownloadKeyModalDialog, {disableClose:true});
-
-                dialogRef.componentInstance.volumeId = res.data.id;
-                dialogRef.componentInstance.fileName = 'pool_' + res.data.name + '_encryption.key';
-                dialogRef.afterClosed().subscribe(result => {
-                  this.goBack();
-                });
-              }
-              else {
-                this.goBack();
+
+        const dialogRef = this.mdDialog.open(EntityJobComponent, {
+          data: { title: T("Creating Pool"), disableClose: true }
+        });
+        dialogRef.componentInstance.setCall(this.pk ? this.editCall : this.addCall, [body]);
+        dialogRef.componentInstance.success
+          .pipe(
+            switchMap((r: any) => {
+              if (this.isEncrypted) {
+                const downloadDialogRef = this.mdDialog.open(DownloadKeyModalDialog, { disableClose: true });
+                downloadDialogRef.componentInstance.volumeId = r.data.id;
+                downloadDialogRef.componentInstance.fileName = "pool_" + r.data.name + "_encryption.key";
+
+                return downloadDialogRef.afterClosed();
               }
-            },
-            (res) => {
-              this.loader.close();
-              if (res.code == 409) {
-                this.error = '';
-                for (let i in res.error) {
-                  res.error[i].forEach(
-                    (error) => { this.error += error + '<br />'; });
+
+              return of(true);
+            }),
+            take(1)
+          )
+          .subscribe(
+            () => {},
+            e => {
+              if (e.code === 409) {
+                this.error = "";
+                for (let i in e.error) {
+                  e.error[i].forEach(error => {
+                    this.error += error + "<br />";
+                  });
                 }
               } else {
-                this.dialog.errorReport(T('Error creating pool'), res.error.error_message, res.error.traceback);
+                this.dialog.errorReport(T("Error creating pool"), res.error.error_message, res.error.traceback);
```

Comment:
this error handling code should probably just be another entityUtils().handleWSError now, probably no need for the if else either, websockets are pretty consistent